exchanges/stream/stream_match.go

```diff
@@ -35,21 +36,27 @@ func (m *Match) IncomingWithData(signature any, data []byte) bool {
 	if !ok {
 		return false
 	}
-	ch <- data
-	close(ch)
-	delete(m.m, signature)
+	ch.waitingRoutine <- data
+	ch.count--
+	if ch.count == 0 {
+		close(ch.waitingRoutine)
+		delete(m.m, signature)
+	}
 	return true
 }
 
 // Set the signature response channel for incoming data
-func (m *Match) Set(signature any) (<-chan []byte, error) {
+func (m *Match) Set(signature any, bufSize int) (<-chan []byte, error) {
```

Comment:
I was thinking we might want to uint this to avoid the below

Reply:
You can't call it with a buf size of 0.
It'd mean you're asking for a blocking channel, right?
We wouldn't want someone to be able to do that would we? ( I really don't know. I think we don't want them to ).
Assuming that, we should error on values of <= 0 and we can accept ints